### PR TITLE
Ports the dictionary lookup feature to the fcitx5 version

### PR DESCRIPTION
**Pull request by zonble**
As the implementation on macOS, a user now can look up a phrase:

- by pressing the "?" key when selecting a candidate in the candidate window.
- by pressing the "?" key when marked a range of the composing buffer.

### CODE REVIEW

**Review by lukhnos — COMMENTED** [review]
After this PR we will have two dependecies which the last public release (2.5.2) didn't have: libicu-dev and libcjson-dev. Previously we only depend on everything that fcitx5 depends (e.g. libfmt-dev). Could this be a problem for potential package maintainers?

For example we would already have to update our Flatpak settings if we were to build against current HEAD. On the other hand, we can also let the Flatpak distribution (or other dist packages) stay at 2.5.2. I'm just wondering what the increased number of deps mean for development velocity (as we want more features) and ease of distribution. Thoughts?

File: CMakeLists.txt [hunk 1]
```diff
@@ -9,6 +9,10 @@ find_package(Fcitx5Utils REQUIRED)
 find_package(fmt REQUIRED)
 find_package(Gettext REQUIRED)
 
+# McBopomofo depends on cJSON. However, not every Linux distro has a newer version contains
+# cmake config files, and we just skip the check.
+#find_package(cJSON REQUIRED)
```
[the remark this answers is not in the document]
Comment (zonble) [reply CMakeLists.txt:14]: I guess another solution is to put a copy of FindcJson.cmake in our repo.

[the remark this answers is not in the document]
Comment (zonble) [reply CMakeLists.txt:14]: I am also considering if there is a format of config file that the fcitx 5 already supports so we can avoid the dependency on a JSON parser.

File: .github/workflows/ci.yaml [hunk 2]
```diff
@@ -45,7 +46,7 @@ jobs:
           apt install -y clang
           apt install -y cmake extra-cmake-modules gettext libfmt-dev
           apt install -y fcitx5 libfcitx5core-dev libfcitx5config-dev libfcitx5utils-dev
-          apt install -y libicu-dev
+          apt install -y libicu-dev libcjson-dev
```
[the remark this answers is not in the document]
Comment (zonble) [reply .github/workflows/ci.yaml]: Updated.

File: CMakeLists.txt (hunk 1 above)
[the remark this answers is not in the document]
Comment (zonble) [reply CMakeLists.txt]: I use json-c instead of cjson. Fcitx5 already has a copy of json-c.

**Review by lukhnos — APPROVED** [review]
Thanks!

**Pull request merged**